### DISCUSSION TITLE
O open-source pode chegar a ser prejudicial para desenvolvedores?

### DISCUSSION THREAD

**Discussion by synga** [frontendbr/forum#1139] [Discussão]
Fala meusa migo, tudo bom?
Seguinte, li hoje o texto [Você é suficiente pra você?](https://medium.com/@isabellasilveira/voc%C3%AA-%C3%A9-suficiente-pra-voc%C3%AA-809a0d35516b) da [@bella-silveira](https://github.com/bella-silveira) (muito om por sina, recomendo a todo dev que puder ler) e em um determinado momento ela fala sobre open-source e a comunidade e disse algo que não havia parado pra pensar, vou copiar aqui um pedaço do texto:

> [...]O problema é que isso virou quase uma cobrança no meio do desenvolvimento, criando uma atmosfera tóxica de disputa de ego e auto importância.[...]

Eu sou relativamente novo nisso de contribuir, o pouco contato que tenho com devs é seguindo-os no Twitter, aqui no Git e Medium, e agora que vou começar a correr atrás de um trampo novo e ter mais contato com isso, sendo assim não vi nenhuma cobrança desse tipo tanto pra cima de mim quanto de outras pessoas. Porém vi que cresceu muito a quantidade de devs indo atrás de talks, meetups, contribuir pra projetos, falando disso, se preocupando com isso.

Pode chegar um momento em que contribuir, palestrar e tudo mais seja um requisito? Por exemplo empresas e eventos selecionando pessoas pela quantidade de commits, variedade de projetos que contribuem, importância das contribuições e etc? É algo que devs novos devem começar a se preocupar e adicionar nas listas de TO DO? Vocês vêem isso crescer a ponto de chegar a excluir pessoas de oportunidades?

Sei que a questão de contratação não está na mão da maioria dos devs, um ou outro da comunidade deve ser alguma especie de líder que faz a avaliação técnica, mas não deveríamos já começar a colocar essa cultura como "open-source é bom, mas não é uma necessidade"?

Enfim, gostaria de saber a opinião de pessoas que estão mais nesse meio de open-source.

**Comment by vinicius73** [comment]
Depende do que definirmos como prejudicial, há sim situações onde open-source (como qualquer projeto na real) pode gerar muito desgaste. [moment().endOf(‘term’)](https://medium.com/@timrwood/moment-endof-term-522d8965689), há outros exemplos similares.

Sobre contratação, palestras e tudo mais...
Bem, como apontado pelo @woliveiras em [A diferença entre ajudar comunidades, ganhar dinheiro com o mercado de educação e o marketing pessoal](https://medium.com/trainingcenter/a-diferença-entre-ajudar-comunidades-ganhar-dinheiro-com-o-mercado-de-educação-e-o-marketing-b8d6b0a85c6b) existem casos e casos...
É inegável que ser ativo em comunidades, projetos open-source, eventos... é um diferencial, mas só pode ser isso nada mais.

Possuímos profissionais incríveis que não necessitosamente estão inseridos na comunidade, assim como possuímos profissionais ativos que não são tão bons assim.
Ser ativo ajuda a *romper* algumas barreiras

Recentemente, no último podcast da Lambda3 [Lambda3 Podcast 101 – Soft skills](https://www.lambda3.com.br/2018/06/lambda3-podcast-101-soft-skills/) falaram muito sobre as skills que se esperam de um profissional atualmente, não lembro de terem comentado sobre atuação em comunidades e projetos open-source.

Existe ainda toda uma gama de marketing pessoal envolvendo isso.  
O importante no final é fazer as coisas com vontade, dedicação e gosto.
Ingressar na área de desenvolvimento nunca foi tão fácil, mas também nunca foi tão difícil. ([Desenvolvedores: Vocês Estão Sozinhos!](https://blog.codecasts.com.br/desenvolvedores-voces-estao-sozinhos-13f3eb846a99)), estar perto da comunidade trocando ideia em locais como este foram é uma maneira de não ficar louco com este mar de informação e novidades.

**Comment by felipefialho** [comment]
Ainda sobre esse assunto, tivemos (@bella-silveira, @eduardojmatos, @raymonsanches e eu) uma resenha absurda de boa no [Dev Na Estrada](https://devnaestrada.com.br/2018/06/22/a-lenda-do-workaholic.html), vale muito escutar.

Vou dar meus dois centavos aqui daqui a pouco.

**Comment by doug2k1** [comment]
Meus R$0,02

Eu acho que começa a ser prejudicial se você faz essas coisas apenas com o pensamento de: 

- fazer pra melhorar currículo (e melhorar o azulejo do GitHub)
- fazer pra impressionar outros devs
- fazer pra ficar famoso

Acho que o mindset mais saudável pra começar um projeto paralelo, uma contribuição no open-source ou preparar uma talk são coisas como:

- fazer para aprender algo novo
- fazer para explorar ideias diferentes
- fazer para resolver um problema de alguém (ou seu mesmo)
- fazer por gratidão, para devolver para a comunidade um pouco do que ela te proporciona

Se você faz com gosto, sem se sentir pressionado (já basta pressão no trabalho), as coisas boas vem por tabela (aprende mais coisas, melhora currículo, ganha mais visibilidade) e evita frustrações.

**Comment by kazzkiq** [comment]
Deve ser consenso que o meio open-source traz várias vantagens profissionais sim. Mas eu acho que isso deve ser tratado como **consequência**, e não **causa** da nossa atuação no open-source.

Se você parar pra pensar, toda a nossa experiência profissional até aqui (no caso da galera iniciante, a partir daqui) passou por projetos open-source. Frameworks, padrões, aquela lib super simples que salvou seu pescoço pra entregar o projeto dentro do prazo... todas essas coisas só foram possíveis porque alguém, um dia, passou por um problema similar ao seu e resolveu compartilhar a solução, de graça.

E se não fosse por essa soma de pessoas compartilhando soluções, nossa vida seria tão mais difícil que dá medo até de imaginar. Por isso acho muito legal quando a gente contribui de volta pro meio open-source, seja compartilhando nossas próprias soluções, ou ajudando na solução de outras pessoas.

Esse espirito de contribuição eventualmente pode te trazer frutos, já que os códigos que você escreve ficam disponíveis publicamente e servem como portfolio. Mas isso é uma coisa que vem naturalmente. Se você contribui, eventualmente vai ter visibilidade e pode receber contato de empresas interessadas. Ou não. Não verdade isso não importa. Acredito que ninguém deveria contribuir apenas pra ganhar visibilidade. Devemos contribuir para retribuir para os outros o que um dia fizeram por nós. Acredito que esse deva ser o espirito.

Se existe panelinha, ou pessoas "estrelinhas" que fazem só pra aparecer, isso é com elas. Eu vou continuar contribuindo como puder, com reconhecimento ou não, pelos motivos que descrevi acima, e acredito que esse seja o pensamento de muitas pessoas no meio open-source.

**Comment by synga** [comment]
Agradeço a resposta de todos, sanaram minhas duvidas. É legal ver que a comunidade mesmo está tentando não criar uma cultura ruim em cima disso. Entendo que pode ser prejudicial dependendo da intenção da pessoa com o open source e alguns outros fatores ai espalhados pelas respostas.

Obrigado @vinicius73 pelas referencias, só falta ler o ultimo, mas adorei o podcast sobre soft skills, já era algo que eu estava de olho desde a issue #911.

@LFeh Cara seus dois centavos foram dados no cast hahahaha já foi bacana ver uma discussão sobre o tema.
